gnd/src/main/java/com/google/android/gnd/repository/DataRepository.java

```diff
@@ -114,51 +159,21 @@ private Completable updateLocalFeature(RemoteDataEvent<Feature> event) {
    * Returns a stream that emits the latest project activation state, and continues to emits changes
    * to that state until all subscriptions are disposed.
    */
-  public Flowable<Persistable<Project>> getActiveProjectOnceAndStream() {
-    return activeProject;
+  public Flowable<Loadable<Project>> getActiveProjectOnceAndStream() {
+    return activeProjectStream;
   }
 
-  public Single<Project> activateProject(String projectId) {
-    Log.d(TAG, " Activating project " + projectId);
-    return remoteDataStore
-        .loadProject(projectId)
-        .flatMap(project -> localDataStore.insertOrUpdateProject(project).toSingleDefault(project))
-        .onErrorResumeNext(
-            throwable -> {
-              if (throwable instanceof FirebaseFirestoreException) {
-                return localDataStore
-                    .getProjectById(localValueStore.getLastActiveProjectId())
-                    .toSingle();
-              }
-              return Single.error(throwable);
-            })
-        .doOnError(throwable -> Log.e(TAG, "Project not found " + projectId))
-        .doOnSubscribe(__ -> activeProject.onNext(Persistable.loading()))
-        .flatMap(project -> localDataStore.insertOrUpdateProject(project).toSingleDefault(project))
-        .doOnSuccess(this::onProjectLoaded);
-  }
-
-  private void onProjectLoaded(Project project) {
-    cache.setActiveProject(project);
-    activeProject.onNext(Persistable.loaded(project));
-    localValueStore.setLastActiveProjectId(project.getId());
+  public void activateProject(String projectId) {
+    Log.v(TAG, "activateProject() called with " + projectId);
+    activateProjectRequests.onNext(Optional.of(projectId));
   }
 
-  public Observable<Persistable<List<Project>>> getProjectSummaries(User user) {
-    // TODO: Get from load db if network connection not available or remote times out.
+  public Flowable<Loadable<List<Project>>> getProjectSummaries(User user) {
     return remoteDataStore
         .loadProjectSummaries(user)
-        .onErrorResumeNext(
-            throwable -> {
-              if (throwable instanceof FirebaseFirestoreException) {
-                return localDataStore.getProjects();
-              }
-              return Single.error(throwable);
-            })
-        .map(Persistable::loaded)
-        .onErrorReturn(Persistable::error)
-        .toObservable()
-        .startWith(Persistable.loading());
+        .onErrorResumeNext(localDataStore.getProjects())
```

Comment:
Same as above

Reply:
I've updated these to log failures so we can track which exceptions are relevant and update accordingly.